### PR TITLE
[AVFoundation] Update AVFoundation to Xcode8-GM

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

File: src/AVFoundation/AVCaptureDeviceDiscoverySession.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#if IOS
+	public partial class AVCaptureDeviceDiscoverySession {
+		[iOS (10,0)]
+		public static AVCaptureDeviceDiscoverySession Create (AVCaptureDeviceType deviceTypes, string mediaType, AVCaptureDevicePosition position)
+		{
+			var deviceTypesArr = Enum.GetValues (typeof (AVCaptureDeviceType))
+				.Cast<AVCaptureDeviceType> ()
+				.Where (f => deviceTypes.HasFlag (f))
+				.Select (o => o.GetConstant ())
+				.ToArray ();
```
Comment (spouliot) [inline src/AVFoundation/AVCaptureDeviceDiscoverySession.cs]: no LINQ in SDK

Comment (dalexsoto) [inline src/AVFoundation/AVCaptureDeviceDiscoverySession.cs]: Oh did not know, changing it

File: src/avfoundation.cs [hunk 2]
```diff
@@ -8216,6 +8216,37 @@ interface AVCaptureStillImageOutput {
...
+		[Static]
+		[Export ("discoverySessionWithDeviceTypes:mediaType:position:")]
+		AVCaptureDeviceDiscoverySession _Create (NSString [] deviceTypes, [NullAllowed] string mediaType, AVCaptureDevicePosition position);
+
+		[Export ("devices")]
+		AVCaptureDevice [] Devices { get; }
+	}
+
+	[NoTV, iOS (10,0), NoMac, NoWatch]
+	[Flags]
```
Comment (spouliot) [inline src/avfoundation.cs]: did you check the generated code ?
I'm not sure I considered flags when dealing with constants...

File: src/avfoundation.cs [hunk 3]
```diff
@@ -8442,6 +8475,35 @@ interface AVCaptureDevice {
...
+		[Internal]
+		[iOS (10,0), NoMac]
+		[Static]
+		[Export ("defaultDeviceWithDeviceType:mediaType:position:")]
+		AVCaptureDevice _DefaultDeviceWithDeviceType (NSString deviceType, string mediaType, AVCaptureDevicePosition position);
+
+		[iOS (10,0), NoMac]
+		[Static]
+		[Wrap ("AVCaptureDevice._DefaultDeviceWithDeviceType (deviceType.GetConstant (), mediaType, position)")]
+		AVCaptureDevice GetDefaultDevice (AVCaptureDeviceType deviceType, string mediaType, AVCaptureDevicePosition position);
```
Comment (spouliot) [inline src/avfoundation.cs]: What happens as `AVCaptureDeviceType` is a `[Flag]` enum ?
`GetConstant` does not deal with that - even if it did the underlying API would not accept an array

File: src/avfoundation.cs [hunk 4]
```diff
@@ -8216,6 +8216,37 @@ interface AVCaptureStillImageOutput {
 #endif
 	}
 
+	[NoTV, iOS (10,0), NoMac, NoWatch]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
```
Comment (spouliot) [inline src/avfoundation.cs]: To ease *future* review(s) add `// init NS_UNAVAILABLE`
That makes it clear it was Apple intent

File: src/avfoundation.cs [hunk 5]
```diff
@@ -8442,6 +8475,35 @@ interface AVCaptureDevice {
...
+		bool LockingWhiteBalanceWithCustomDeviceGainsSupported { [Bind ("isLockingWhiteBalanceWithCustomDeviceGainsSupported")] get; }
+
+		// From AVCaptureDevice (AVCaptureDeviceType) Category
+		[Internal]
+		[iOS (10,0), NoMac]
+		[Export ("deviceType")]
+		NSString _DeviceType { get; }
+
+		[iOS (10, 0), NoMac]
+		[Wrap ("AVCaptureDeviceTypeExtensions.GetValue (_DeviceType)")]
```
Comment (spouliot) [inline src/avfoundation.cs]: nice :)

**Comment by spouliot** [comment]
At first glance the flags made sense - but the API usage (both current and the *unknown* future) makes it problematic. Let's stick to a normal `enum` as it solve both issues :)

**Comment by dalexsoto** [comment]
@spouliot fixed 😄

**Comment by dalexsoto** [comment]
build

**Comment by spouliot** [comment]
LGTM

**Pull request merged**